compiler/x/codegen/FPTreeEvaluator.cpp

```diff
@@ -524,12 +524,18 @@ TR::Register *OMR::X86::TreeEvaluator::fpUnaryMaskEvaluator(TR::Node *node, TR::
       result->setIsSinglePrecision();
       }
 
-   // TODO 3-OP Optimization
-   if (result != value)
+   if (cg->comp()->target().cpu.supportsAVX())
       {
-      generateRegRegInstruction(TR::InstOpCode::MOVDQURegReg, node, result, value, cg);
+      generateRegRegMemInstruction(opcode, node, result, value, generateX86MemoryReference(cg->findOrCreate16ByteConstant(node, mask), cg), cg);
```

Comment:
To share some .so code space, can you hoist the `generateX86MemoryReference(cg->findOrCreate16ByteConstant(node, mask), cg)` above the `if` and share it on both branches?

Reply:
See [force-push](https://github.com/eclipse/omr/compare/9b2b12505e8257c586dbc2f6880f059f9ad2c364..bc9c56a47997673716c3a4e814c2bec93d61b2b0)